src/common/ReactHooks.ts

```diff
@@ -185,107 +186,115 @@ export function defineComponentHook<C extends ReactExtended.WritableComponent>(
 	criteria: ComponentCriteria,
 	options: HookOptions<C> = {},
 ): ReactComponentHook<C> {
+	const retry = () => update();
+
 	const hook: ReactComponentHook<C> = reactive({
 		cls: undefined,
 		instances: [],
 		unhooked: false,
+		retry,
 	});
 
-	const components = awaitComponents(criteria);
+	const update = () => {
+		if (hook.watcher) hook.watcher.disconnect();
```

Comment:
```suggestion
		if (hook.watcher) {
				hook.watcher.disconnect();
				hook.watcher = undefined;
		}
```

Should also set `hook.watcher` to undefined here, since its not guaranteed to be set later on (If the component is found immediately in tree, it will never be unset otherwise, causing memory leakage)